src/edu/csus/ecs/pc2/ui/EditProblemPane.java

```diff
@@ -715,232 +724,258 @@ public void enableUpdateButton() {
                             }
                         }
                     }
+                    
+                } else {
+                    logDebugException("No ProblemDataFiles for " + problem);
+                }
+
+
+                // see if the choice of which output validator (if any) to use has changed;
+                // enable the Update button and update the tooltip if so
+                if (problem.getValidatorType() != changedProblem.getValidatorType()) {
+                    enableButton = true;
+                    if (updateToolTip.equals("")) {
+                        updateToolTip = "Output Validator";
+                    } else {
+                        updateToolTip += ", Output Validator";
+                    }
+                }
 
-                    // see if the choice of which output validator (if any) to use has changed;
```

Comment:
To be honest, the diffs in the next 250 lines of code are so convoluted/complex that I find myself unable to review them.  I _think_ these diffs may have been generated because the diff program itself got out of sync; it _looks_ to me like most of these diffs are merely reshuffling of code due to relatively simple line changes.  But I can't say for certain... I'm not sure what to do regarding reviewing this part; any suggestions or explanations as to what "really changed"?

Reply:
In github, if you switch to the branch that contains the file, and select "blame", as in this link: [https://github.com/johnbrvc/pc2v9/blame/i_295_sandbox/src/edu/csus/ecs/pc2/ui/EditProblemPane.java](https://github.com/johnbrvc/pc2v9/blame/i_295_sandbox/src/edu/csus/ecs/pc2/ui/EditProblemPane.java) you can see the individual changes and comments in the left pane (scroll down to around line 700).  You can click on one of those comments to see the changes specific to that comment.  A lot has to do with indenting and refactoring of existing code you must have done awhile ago on that i295 branch.

Reply:
Ok, yep; I reviewed that and it appears to me that nearly all the diffs are due to the development I was doing last year.  I'm going to go out on a limb and assume that either (a) everything is fine or else (b) problems related to this GUI class will show up fairly quickly.  I hope :)